web/src/user_group_edit.js

```diff
@@ -362,7 +366,77 @@ function update_toggler_for_group_setting() {
     toggler.goto(select_tab);
 }
 
+function get_subgroup_names_for_status_text(subgroups) {
+    const subgroup_names = subgroups.map((subgroup) => subgroup.name);
+    if (Intl.ListFormat === undefined) {
+        return subgroup_names
+            .map((name) => `<b>${Handlebars.Utils.escapeExpression(name)}</b>`)
+            .join(", ");
+    }
+
+    const list_formatter = new Intl.ListFormat(user_settings.default_language, {
+        style: "long",
+        type: "unit",
+    });
+    const formatted_parts = list_formatter.formatToParts(subgroup_names);
+    return formatted_parts
+        .map((part) => {
+            if (part.type === "element") {
+                return `<b>${Handlebars.Utils.escapeExpression(part.value)}</b>`;
+            }
+            return part.value;
+        })
+        .join("");
+}
+
+function update_membership_status_text(group) {
+    const current_user_id = people.my_current_user_id();
+    const is_direct_member = user_groups.is_direct_member_of(current_user_id, group.id);
+
+    let is_member;
+    let associated_subgroup_names_string;
+    if (is_direct_member) {
+        is_member = true;
+    } else {
+        is_member = user_groups.is_user_in_group(group.id, current_user_id);
+        if (is_member) {
+            const associated_subgroups = user_groups.get_associated_subgroups(
+                group,
+                current_user_id,
+            );
+            associated_subgroup_names_string =
+                get_subgroup_names_for_status_text(associated_subgroups);
+        }
+    }
```

Comment:
I feel like the calculation of the variables to pass into `render_user_group_membership_status` could perhaps be a shared function used by both code paths.

Reply:
Added `get_membership_status_context`.